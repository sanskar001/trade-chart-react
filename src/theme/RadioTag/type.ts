type ChangeHandlerType = React.ChangeEventHandler<HTMLInputElement>;

export interface RadioTagProps {
  value: string;
  name: string;
  checked?: boolean;
  onChange?: ChangeHandlerType;
  label?: React.ReactNode;
  className?: string;
  style?: React.CSSProperties;
  disabled?: boolean;
}
