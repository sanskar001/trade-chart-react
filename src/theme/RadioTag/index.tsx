import React from "react";
import { formatClass } from "../../util/formatClass";
import { RadioTagProps } from "./type";

const RadioTag: React.FC<RadioTagProps> = ({
  value,
  label,
  checked = false,
  onChange,
  name = "radio",
  className = "",
  style,
  disabled = false,
}) => {
  return (
    <div className={formatClass(`radio-tag ${className}`)} style={style}>
      <input
        type="radio"
        id={value}
        name={name}
        value={value}
        checked={checked}
        onChange={onChange}
        disabled={disabled}
      />
      <label htmlFor={value}>{label || value}</label>
    </div>
  );
};

export default RadioTag;
