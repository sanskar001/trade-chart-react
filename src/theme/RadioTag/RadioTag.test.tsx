import renderer from "react-test-renderer";
import RadioTag from ".";

const changeHandler = jest.fn();

it("renders correctly when there is default radioTag", () => {
  const tree = renderer
    .create(<RadioTag name="radio-filter" value="first" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is radioTag with onChange event handle", () => {
  const tree = renderer
    .create(
      <RadioTag name="radio-filter" value="first" onChange={changeHandler} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is checked radioTag", () => {
  const tree = renderer
    .create(
      <RadioTag
        name="radio-filter"
        value="first"
        checked
        onChange={changeHandler}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is disabled radioTag", () => {
  const tree = renderer
    .create(
      <RadioTag
        name="radio-filter"
        value="first"
        disabled
        onChange={changeHandler}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is radioTag with label", () => {
  const tree = renderer
    .create(
      <RadioTag
        name="radio-filter"
        value="first"
        label={<h4>First</h4>}
        onChange={changeHandler}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is radioTag with custom style and class", () => {
  const tree = renderer
    .create(
      <RadioTag
        name="radio-filter"
        value="first"
        label={<h4>First</h4>}
        onChange={changeHandler}
        className="custom-radio"
        style={{ border: "2px solid yellow" }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
