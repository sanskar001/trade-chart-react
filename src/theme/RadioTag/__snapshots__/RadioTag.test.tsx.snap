// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is checked radioTag 1`] = `
<div
  className="radio-tag"
>
  <input
    checked={true}
    disabled={false}
    id="first"
    name="radio-filter"
    onChange={[MockFunction]}
    type="radio"
    value="first"
  />
  <label
    htmlFor="first"
  >
    first
  </label>
</div>
`;

exports[`renders correctly when there is default radioTag 1`] = `
<div
  className="radio-tag"
>
  <input
    checked={false}
    disabled={false}
    id="first"
    name="radio-filter"
    type="radio"
    value="first"
  />
  <label
    htmlFor="first"
  >
    first
  </label>
</div>
`;

exports[`renders correctly when there is disabled radioTag 1`] = `
<div
  className="radio-tag"
>
  <input
    checked={false}
    disabled={true}
    id="first"
    name="radio-filter"
    onChange={[MockFunction]}
    type="radio"
    value="first"
  />
  <label
    htmlFor="first"
  >
    first
  </label>
</div>
`;

exports[`renders correctly when there is radioTag with custom style and class 1`] = `
<div
  className="radio-tag custom-radio"
  style={
    Object {
      "border": "2px solid yellow",
    }
  }
>
  <input
    checked={false}
    disabled={false}
    id="first"
    name="radio-filter"
    onChange={[MockFunction]}
    type="radio"
    value="first"
  />
  <label
    htmlFor="first"
  >
    <h4>
      First
    </h4>
  </label>
</div>
`;

exports[`renders correctly when there is radioTag with label 1`] = `
<div
  className="radio-tag"
>
  <input
    checked={false}
    disabled={false}
    id="first"
    name="radio-filter"
    onChange={[MockFunction]}
    type="radio"
    value="first"
  />
  <label
    htmlFor="first"
  >
    <h4>
      First
    </h4>
  </label>
</div>
`;

exports[`renders correctly when there is radioTag with onChange event handle 1`] = `
<div
  className="radio-tag"
>
  <input
    checked={false}
    disabled={false}
    id="first"
    name="radio-filter"
    onChange={[MockFunction]}
    type="radio"
    value="first"
  />
  <label
    htmlFor="first"
  >
    first
  </label>
</div>
`;
